src/api/events.ts

```diff
@@ -1,9 +1,44 @@
 import { Event } from '../types/domain';
 
+let EVENTS_CACHE: Array<Event> | null = null;
+const EVENT_CACHE: Record<string, Event> = {};
+
 export async function fetchEvents(): Promise<Array<Event>> {
+  if (EVENTS_CACHE != null) {
+    return Promise.resolve(EVENTS_CACHE);
+  }
+
   const response = await fetch('https://localhost:5001/api/event');
   if (response.ok) {
-    return await response.json();
+    const events = await response.json();
+    cacheEvents(events);
+    return events;
   }
   throw new Error(`Failed to fetch events: ${response.statusText}.`);
 }
+
+export async function fetchEvent(id: string): Promise<Event> {
+  const cachedEvent = EVENT_CACHE[id];
+  if (cachedEvent != null) {
+    return Promise.resolve(cachedEvent);
+  }
+
+  const response = await fetch(`https://localhost:5001/api/event/${id}`);
```

Comment:
Maybe the base URL here should be a config variable or from a `.env` file or something now that it's used more places?

Reply:
I agree, but this can be done in a separate PR - maybe the one that addresses #5.